src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -673,14 +673,8 @@ private static BoundStatement BindStatement(Binder binder, StatementSyntax synta
 
         private static bool IsAssignableExpression(Binder binder, BoundExpression expression)
         {
-            //PROTOTYPE(readonlyRefs): is this comment still valid?
-            // NOTE: Surprisingly, binder.CheckValueKind will return true (!) for readonly fields 
-            // in contexts where they cannot be assigned - it simply reports a diagnostic.
-            // Presumably, this is done to avoid producing a confusing error message about the
-            // field not being an lvalue.
             var diagnostics = DiagnosticBag.GetInstance();
-            var result = binder.CheckValueKind(expression.Syntax, expression, Binder.BindValueKind.Assignable, checkingReceiver: false, diagnostics: diagnostics) &&
-                !diagnostics.HasAnyErrors();
```

Comment:
There doesn't seem to be any EE tests affected. Is this covered?

Reply:
The behavior that comment describes is no longer happening. CheckValueKind will never return `true` if it returns diagnostics. The workaround was needed because in the past you could not request "Assignable" and CheckValueKind would produce `true` for readonly fields since it did not know whether you want read or write.

The workaround is no longer needed. This is also the only place where such workaround was used.

Removing the workaround used to cause failures with the old version of CheckValueKind. Evidently nothing fails now.

---
In reply to: [139489173](https://github.com/dotnet/roslyn/pull/21982#discussion_r139489173) [](ancestors = 139489173)

Reply:
Makes sense. Thanks